### PR TITLE
Make changes to conditionally create dapr_default_config subchart

### PR DESCRIPTION
**Pull request by pruthvidhodda**
# Description

Made changes to conditionally create dapr_default_config subchart

## Issue reference

We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: [#_[1213]_](https://github.com/dapr/dapr/issues/1213)

## Checklist

Please make sure you've  completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [ ] Created/updated tests
* [x] Unit tests passing
* [ ] End-to-end tests passing
* [ ] Extended the documentation
* [ ] Specification has been updated
* [ ] Provided sample for the feature

### CODE REVIEW

**Comment by yaron2** [comment]
cc @youngbupark

File: charts/dapr/values.yaml [hunk 1]
```diff
@@ -6,4 +6,7 @@ global:
   imagePullSecrets: ""
   prometheus:
     enabled: true
-    port: 9090
\ No newline at end of file
+    port: 9090
+
+dapr_config:
+  enabled: true
```
[the remark this answers is not in the document]
Comment (pruthvidhodda) [reply charts/dapr/values.yaml]: Global values are accessible from all sub-charts but we want this value to be accessed by dapr_config only. So isn't it better this way?

[the remark this answers is not in the document]
Comment (youngbupark) [reply charts/dapr/values.yaml]: Ah ok. then that make sense.

[the remark this answers is not in the document]
Comment (youngbupark) [reply charts/dapr/values.yaml]: You need to move enabled to ./charts/dapr/dapr_config/values.yaml  instead of putting this in value of umbrella chart.

**Comment by yaron2** [comment]
So just to understand, how does this PR work towards the CLI error?

**Comment by yaron2** [comment]
I wouldn't allow people to remove the default config, it could cause undesirable behaviors. We have it mentioned in the docs..

If a user wants to edit or change the config they can do that with Helm, but why would the user want to remove it completely?

**Comment by youngbupark** [comment]
> I wouldn't allow people to remove the default config, it could cause undesirable behaviors. We have it mentioned in the docs..
> 
> If a user wants to edit or change the config they can do that with Helm, but why would the user want to remove it completely?

yaml will be generated during release process instead of hand-crafting yaml to remove config for cli. 

No this is not for user. this is for our build process. we will move this option to subchart. we can add some comment around in value.yaml 

We could not find the better way than this.

**Comment by yaron2** [comment]
> > I wouldn't allow people to remove the default config, it could cause undesirable behaviors. We have it mentioned in the docs..
> > If a user wants to edit or change the config they can do that with Helm, but why would the user want to remove it completely?
> 
> yaml will be generated during release process instead of hand-crafting yaml to remove config for cli.
> 
> No this is not for user. this is for our build process. we will move this option to subchart. we can add some comment around in value.yaml
> 
> We could not find the better way than this.

I will not object to this, but I don't like giving users options by which they can ruin their environments..

**Comment by youngbupark** [comment]
> I will not object to this, but I don't like giving users options by which they can ruin their environments..

People do not know as long as they look at the sub chart values. We will add comment to values.yaml

File: charts/dapr/values.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (pruthvidhodda) [reply charts/dapr/values.yaml]: Done.

File: charts/dapr/charts/dapr_config/values.yaml [hunk 2]
```diff
@@ -1,4 +1,6 @@
 mtls:
   enabled: "true"
   workloadCertTTL: 24h
-  allowedClockSkew: 15m
\ No newline at end of file
+  allowedClockSkew: 15m
+
+  enabled: true
```
Comment (youngbupark) [inline charts/dapr/charts/dapr_config/values.yaml]: Can we rename `enabled` to be more meaningful ? e.g. `config_chart_included`

And please add the purpose of this value as a comment.

File: charts/dapr/charts/dapr_config/values.yaml [hunk 3]
```diff
@@ -1,4 +1,7 @@
 mtls:
   enabled: "true"
   workloadCertTTL: 24h
-  allowedClockSkew: 15m
\ No newline at end of file
+  allowedClockSkew: 15m
+
+  # This value is essential in determining whether to build this chart or not based on the usecase. For example this chart should not be included while generating manifest file for dapr which is achieved by overriding this value.
+  dapr_config_chart_included: true
```
Comment (youngbupark) [inline charts/dapr/charts/dapr_config/values.yaml]: nit: please remove tab spaces.

**Comment by youngbupark** [comment]
for validation,
0. Create new minikube cluster
1. generate manifest yaml
2. kubectl apply -f ./dist/install/dapr.yaml

Please make sure that manifest yaml is working as expected.

**Comment by pruthvidhodda** [comment]
Added a flag to include CRD's in templating. Verified locally on my minikube cluster.

**Pull request merged**